storage/system-test/transfer.test.js

```diff
@@ -0,0 +1,125 @@
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var uuid = require('node-uuid');
+var program = require('../transfer');
+var Storage = require('@google-cloud/storage');
+
+var storage = Storage();
+var bucketName = 'nodejs-docs-samples-test-' + uuid.v4();
+var bucketName2 = 'nodejs-docs-samples-test-' + uuid.v4();
+
+describe.only('storage:transfer', function () {
+  var jobName;
+  var date = '2222/08/11';
+  var time = '15:30';
+  var description = 'this is a test';
+  var status = 'DISABLED';
+
+  before(function (done) {
+    storage.createBucket(bucketName, function (err) {
+      if (err) {
+        return done(err);
+      }
+      storage.createBucket(bucketName2, done);
+    });
+  });
+
+  after(function (done) {
+    storage.bucket(bucketName).deleteFiles({ force: true }, function (err) {
+      if (err) {
+        return done(err);
+      }
+      storage.bucket(bucketName).delete(function (err) {
+        if (err) {
+          return done(err);
+        }
+        storage.bucket(bucketName2).deleteFiles({ force: true }, function (err) {
+          if (err) {
+            return done(err);
+          }
+          storage.bucket(bucketName2).delete(done);
+        });
+      });
+    });
+  });
+
+  describe('createTransferJob', function () {
+    it('should create a storage transfer job', function (done) {
+      program.createTransferJob(bucketName, bucketName2, date, time, description, function (err, transferJob) {
+        assert.ifError(err);
+        jobName = transferJob.name;
+        assert.equal(transferJob.name.indexOf('transferJobs/'), 0);
+        assert.equal(transferJob.description, description);
+        assert.equal(transferJob.status, 'ENABLED');
+        assert(console.log.calledWith('Created transfer job: %s', transferJob.name));
+        setTimeout(done, 2000);
+      });
+    });
+  });
+
+  describe('getTransferJob', function () {
+    it('should get a transferJob', function (done) {
+      program.getTransferJob(jobName, function (err, transferJob) {
+        assert.ifError(err);
+        assert.equal(transferJob.name, jobName);
+        assert.equal(transferJob.description, description);
+        assert.equal(transferJob.status, 'ENABLED');
+        assert(console.log.calledWith('Found transfer job: %s', transferJob.name));
+        setTimeout(done, 2000);
+      });
+    });
+  });
+
+  describe('updateTransferJob', function () {
+    it('should update a transferJob', function (done) {
+      program.updateTransferJob(jobName, 'status', status, function (err, transferJob) {
+        assert.ifError(err);
+        assert.equal(transferJob.name, jobName);
```

Comment:
Nah, I don't want to validate that the Storage Transfer API is responding with all of the correct fields. Ideally they have their own tests for that. This assertion is sufficient to say yes, the sample used the client library to hit a service and we got a response back the _looks_ okay.

Now, if the sample itself were using some other fields, then maybe we'd want to verify them.